lib/semian/net_http.rb

```diff
@@ -0,0 +1,89 @@
+require 'semian'
+require 'semian/adapter'
+
+module Net
+  ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  ResourceBusyError = Class.new(SemianError)
+  CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module NetHTTP
+    include Semian::Adapter
+
+    ResourceBusyError = ::Net::ResourceBusyError
+    CircuitOpenError = ::Net::CircuitOpenError
+
+    def semian_identifier
+      Semian::NetHTTP.retrieve_semian_configuration(address, port)[:identifier]
```

Comment:
I still feel like we should use `:name`, add a prefix and make it a symbol here. I don't think we should let the user set `semian_identifier` this directly here. Both the mysql and redis adapters already do this and also have a default.